src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -45,25 +44,43 @@ public static T GetSingleValueOrDefault<T>(this DicomDataset dicomDataset, Dicom
             return dicomDataset.GetSingleValueOrDefault<T>(dicomTag, default);
         }
 
+        public static long? GetDicomTagValueAsLong(this DicomDataset dicomDataset, DicomTag dicomTag)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+
+            DicomTag tag = dicomDataset.GetSingleValueOrDefault<DicomTag>(dicomTag, default);
+            if (tag == null)
```

Comment:
Why are you getting a tag here? Isn't the goal to get the long value for the tag?

Reply:
maybe I should rename the "DicomTag tag" to "DicomTag value" to avoid this confusion.
Attribute AT is supposed to store a DicomTag value, we need to convert it to long  to store to databse.

Reply:
Value still maps to the value for a given tag... potentially you should be more specific in tying this whole function to AT. It doesn't clearly tie to that for me. I would say the tag variable should be something like valueForATTag and the function name as well should specify AT.

---
In reply to: [591944229](https://github.com/microsoft/dicom-server/pull/612#discussion_r591944229) [](ancestors = 591944229)

Reply:
may be GetAttributeTagValueAsLong?